go-libp2p test run failed with a timeout error

Seems like an instance of flakiness:
https://github.com/libp2p/go-libp2p/actions/runs/3008550295

```
# run started:
2022-09-07T14:57:59.7033399Z [36;1mtestground run composition                          \[0m

# build (cached) completed
2022-09-07T15:03:45.6342293Z Successfully built 6a9d84d42825
2022-09-07T15:03:45.6344466Z Successfully tagged 1ec9becf2a5b:latest
2022-09-07T15:03:45.6347691Z Sep  7 15:03:45.008629	[34mINFO[0m	build completed	{"default_tag": "1ec9becf2a5b:latest", "took": "5m44s"}
2022-09-07T15:03:45.6348197Z Sep  7 15:03:45.061610	[34mINFO[0m	got docker image id	{"image_id": "6a9d84d42825"}

# containers started
2022-09-07T15:03:45.6382450Z Sep  7 15:03:45.512205	[34mINFO[0m	starting containers	{"runner": "local:docker", "run_id": "cccb2ttbgtls8cq0m27g", "count": 8}
2022-09-07T15:03:45.6383196Z Sep  7 15:03:45.512295	[34mINFO[0m	starting container	{"runner": "local:docker", "run_id": "cccb2ttbgtls8cq0m27g", "id": "558457e509b9719f9492f80330c9c661edd0cc0e858e4a7975a4f46532081a5b", "group": "v0.20", "group_index": 0}
2022-09-07T15:03:45.6384228Z Sep  7 15:03:45.512443	[34mINFO[0m	starting container	{"runner": "local:docker", "run_id": "cccb2ttbgtls8cq0m27g", "id": "b500afb02b1a477d34302770fe24d592fc546109cd0b2d611ee4a5c0e4ab5aaf", "group": "v0.22", "group_index": 0}
2022-09-07T15:03:45.6385039Z Sep  7 15:03:45.512542	[34mINFO[0m	starting container	{"runner": "local:docker", "run_id": "cccb2ttbgtls8cq0m27g", "id": "22207f6159c7d414fd0d317323935e5660ea9ac7b7ee296f9bab7d8718a05a8f", "group": "v0.19", "group_index": 0}
2022-09-07T15:03:45.6386037Z Sep  7 15:03:45.512681	[34mINFO[0m	starting container	{"runner": "local:docker", "run_id": "cccb2ttbgtls8cq0m27g", "id": "e6ecfb7cd580e0629edc59f3f30ef8f49f9eb793a87a3969f8f5ca58b928366b", "group": "v0.17", "group_index": 0}
2022-09-07T15:03:45.6386871Z Sep  7 15:03:45.512774	[34mINFO[0m	starting container	{"runner": "local:docker", "run_id": "cccb2ttbgtls8cq0m27g", "id": "e4a40bb3fea59c4073ac6129b131bf6ea2fc846d8d18e456dbc56364e3ed239a", "group": "v0.11", "group_index": 0}
2022-09-07T15:03:45.6387689Z Sep  7 15:03:45.512876	[34mINFO[0m	starting container	{"runner": "local:docker", "run_id": "cccb2ttbgtls8cq0m27g", "id": "69cf5de57fa2d54c71e62f44c7704a9e16b03dfe102ef5a6967bc521c26e5525", "group": "v0.21", "group_index": 0}
2022-09-07T15:03:45.6388618Z Sep  7 15:03:45.512923	[34mINFO[0m	starting container	{"runner": "local:docker", "run_id": "cccb2ttbgtls8cq0m27g", "id": "911514f5fcd736411b0128b2ee8dc8a1e6ea440b2292553bb50c451be8c34c5a", "group": "master", "group_index": 0}
2022-09-07T15:03:45.6397574Z Sep  7 15:03:45.512941	[34mINFO[0m	starting container	{"runner": "local:docker", "run_id": "cccb2ttbgtls8cq0m27g", "id": "3198996e23b58f6323a1f3c84dcea02c993782f5801930a0caa52abee1a46857", "group": "custom", "group_index": 0}
2022-09-07T15:03:47.6386536Z Sep  7 15:03:47.086694	[34mINFO[0m	attaching container	{"runner": "local:docker", "run_id": "cccb2ttbgtls8cq0m27g", "id": "e4a40bb3fea59c4073ac6129b131bf6ea2fc846d8d18e456dbc56364e3ed239a", "group": "v0.11", "group_index": 0}
2022-09-07T15:03:47.6387482Z Sep  7 15:03:47.148403	[34mINFO[0m	1.6358s [37;45m     OTHER[0m [31m<< sidecar      >>[0m Sep  7 15:03:47.144533	[34mINFO[0m	resolved route to host	{"host": "testground-redis", "ip": "192.18.0.3"}

# run finished
2022-09-07T15:04:02.6647215Z Sep  7 15:04:02.151186	[34mINFO[0m	deleting container	{"runner": "local:docker", "run_id": "cccb2ttbgtls8cq0m27g", "id": "911514f5fcd736411b0128b2ee8dc8a1e6ea440b2292553bb50c451be8c34c5a"}
2022-09-07T15:04:02.6647849Z Sep  7 15:04:02.151355	[34mINFO[0m	16.6388s [37;101mINCOMPLETE[0m [31m<< sidecar      >>[0m 
2022-09-07T15:04:02.6648748Z Sep  7 15:04:02.231300	[34mINFO[0m	run finished with outcome = success (custom:1/1 master:1/1 v0.11:1/1 v0.17:1/1 v0.19:1/1 v0.20:1/1 v0.21:1/1 v0.22:1/1)	{"run_id": "cccb2ttbgtls8cq0m27g", "plan": "ping", "case": "ping", "runner": "local:docker", "instances": 8}
2022-09-07T15:04:04.6655920Z [1;92m
2022-09-07T15:04:04.6656507Z >>> Result:
2022-09-07T15:04:04.6657088Z [0m
2022-09-07T15:04:04.6670036Z Sep  7 15:04:04.666708	[34mINFO[0m	finished run with ID: cccb2ttbgtls8cq0m27g
2022-09-07T15:04:04.6733547Z [1;92m
2022-09-07T15:04:04.6733931Z >>> Result:
2022-09-07T15:04:04.6734169Z [0m
2022-09-07T15:04:04.6734881Z Sep  7 15:04:04.673122	[34mINFO[0m	created file: ./result.tgz
2022-09-07T15:04:04.6805210Z ##[error]The action has timed out.
2022-09-07T15:04:04.6915639Z ##[group]Run actions/upload-artifact@v3
```



raw logs:
[4.txt](https://github.com/libp2p/test-plans/files/9507435/4.txt)

--- a/.github/actions/setup-testground/action.yml
+++ b/.github/actions/setup-testground/action.yml
@@ -19,8 +19,13 @@ runs:
       if: ${{ inputs.testground_ref == 'edge' }}
       shell: bash
       run: |
-        /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/testground/testground/master/install.sh)"
-
+        for i in 1 2 3; do
+          echo "=== Attempt $i ==="
+          /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/testground/testground/master/install.sh)" && \
+            exit 0
+          sleep 10
+        done
+        exit 1
     # Custom setup (slower) when we use a different testground_ref
     - name: Checkout testground
       if: ${{ inputs.testground_ref != 'edge' }}

--- a/.github/workflows/run-composition.yml
+++ b/.github/workflows/run-composition.yml
@@ -36,6 +36,7 @@ jobs:
       TEST_PLAN_REPO: ${{ inputs.test_repository || 'libp2p/test-plans' }}
       TEST_PLAN_BRANCH: ${{ inputs.test_ref || 'master' }}
       TESTGROUND_ENDPOINT: ${{ inputs.testground_endpoint }}
+      COMPOSITION_FILE: ${{ inputs.composition_file }}
     defaults:
       run:
         shell: bash
@@ -67,7 +68,8 @@ jobs:
           for i in 1 2 3; do
             echo "=== Attempt $i ==="
             testground build composition                        \
-              -f ${{ inputs.composition_file }}                 \
+              -f "${COMPOSITION_FILE}"                          \
+              --write-artifacts                                 \
               --wait && exit 0;
             sleep 10
           done
@@ -81,7 +83,7 @@ jobs:
         timeout-minutes: 6
         run: |
           testground run composition                          \
-            -f ${{ inputs.composition_file }}                 \
+            -f "${COMPOSITION_FILE}"                          \
             --metadata-repo "${GITHUB_REPOSITORY}"            \
             --metadata-branch "${GITHUB_REF#refs/heads/}"     \
             --metadata-commit "${GITHUB_SHA}"                 \
@@ -98,5 +100,6 @@ jobs:
           path: |
             ~/testground/
             ~/test-plans/result.tgz
+            ${{env.COMPOSITION_FILE}}
             testground.*
             test-plans/*.out

--- a/.github/workflows/run-composition.yml
+++ b/.github/workflows/run-composition.yml
@@ -63,7 +63,7 @@ jobs:
           echo "::set-output name=custom_git_sha::${SHA}"
       - name: Build the composition file
         working-directory: ./test-plans
-        timeout-minutes: 30
+        timeout-minutes: 40
         run: |
           for i in 1 2 3; do
             echo "=== Attempt $i ==="

